analytics/src/analytics/metrics/percent_complete.py

```diff
@@ -0,0 +1,132 @@
+        self.dataset = dataset
+        super().__init__()
+
+    def calculate(self) -> pd.DataFrame:
+        """
+        Calculate the percent complete per deliverable.
+
+        Notes
+        -----
+        Percent completion is calculated using the following steps:
+        1. Count the number of all tasks (or points) per deliverable
+        2. Count the number of closed tasks (or points) per deliverable
+        3. Left join all tasks/points with open tasks/points on deliverable
+           so that we have a row per deliverable with a total count column
+           and a closed count column
+        4. Subtract closed count from total count to get open count
+        5. Divide closed count by total count to get percent complete
+        """
+        # get total and closed counts per deliverable
+        df_total = self._get_count_by_deliverable(status="all", unit=self.unit)
+        df_closed = self._get_count_by_deliverable(status="closed", unit=self.unit)
+        # join total and closed counts on deliverable
+        # and calculate remaining columns
+        df_all = df_total.merge(df_closed, on=self.deliverable_col, how="left")
+        df_all = df_all.fillna(0)
+        df_all["open"] = df_all["total"] - df_all["closed"]
+        df_all["percent_complete"] = df_all["closed"] / df_all["total"]
+        df_all["percent_complete"] = df_all["percent_complete"].fillna(0)
+        return df_all
+
+    def plot_results(self) -> Figure:
+        """Create a bar chart of percent completion from the data in self.results."""
+        # reshape the dataframe in self.results for plotly
+        df = self._prepare_result_dataframe_for_plotly()
+        # create a stacked bar chart from the data
+        return px.bar(
+            df,
+            x=self.unit,
+            y=self.deliverable_col,
+            color=self.status_col,
+            text="percent_of_total",
+            color_discrete_map={"open": "#aacde3", "closed": "#06508f"},
+            orientation="h",
+            title=f"Deliverable Percent Complete by {self.unit}",
+            height=800,
+        )
+
+    def post_results_to_slack(self, slackbot: SlackBot, channel_id: str) -> None:
+        """Post sprint burndown results and chart to slack channel."""
+        return super()._post_results_to_slack(
+            slackbot=slackbot,
+            channel_id=channel_id,
+            message="*:github: Percent complete by deliverable*",
+        )
+
+    def _get_count_by_deliverable(
+        self,
+        status: str,
+        unit: Literal["tasks", "points"] = "points",
+    ) -> pd.DataFrame:
+        """Get the count of tasks (or points) by deliverable and status."""
+        # create local copies of the dataset and key column names
+        df = self.dataset.df.copy()
+        key_cols = [self.deliverable_col, unit]
+        # create a dummy column to sum per row if the unit is tasks
+        if unit == "tasks":
```

Comment:
should this be using the Unit class? Or am i misunderstanding. 

Same question about line 91.

Reply:
Because I'm using `unit` in this case for column names it was helpful to have it stored as a str. But I was able to do some refactoring and use `unit.value` in most places to avoid hardcoding the string value in most places.

Those updates are in [this commit](https://github.com/HHS/simpler-grants-gov/pull/782/commits/841f141440912f4bbc3a2530bab868c707f2b47f) of #782